entity_api/src/coaching_session.rs

```diff
@@ -0,0 +1,121 @@
+        let db = MockDatabase::new(DatabaseBackend::Postgres).into_connection();
+        let mut query_params = HashMap::new();
+        let coaching_relationship_id = Id::new_v4();
+
+        query_params.insert(
+            "coaching_relationship_id".to_owned(),
+            coaching_relationship_id.to_string(),
+        );
+
+        let _ = find_by(&db, query_params).await;
+
+        assert_eq!(
+            db.into_transaction_log(),
+            [Transaction::from_sql_and_values(
+                DatabaseBackend::Postgres,
+                r#"SELECT "coaching_sessions"."id", "coaching_sessions"."coaching_relationship_id", "coaching_sessions"."date", "coaching_sessions"."timezone", "coaching_sessions"."created_at", "coaching_sessions"."updated_at" FROM "refactor_platform"."coaching_sessions" WHERE "coaching_sessions"."coaching_relationship_id" = $1"#,
+                [coaching_relationship_id.into()]
+            )]
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn find_by_from_date_returns_all_records_after_date() -> Result<(), Error> {
+        let db = MockDatabase::new(DatabaseBackend::Postgres).into_connection();
+        let mut query_params = HashMap::new();
+        let from_date = NaiveDate::from_ymd_opt(2021, 1, 1).unwrap();
+
+        query_params.insert("from_date".to_owned(), from_date.to_string());
+
+        let _ = find_by(&db, query_params).await;
+
+        assert_eq!(
+            db.into_transaction_log(),
+            [Transaction::from_sql_and_values(
+                DatabaseBackend::Postgres,
+                r#"SELECT "coaching_sessions"."id", "coaching_sessions"."coaching_relationship_id", "coaching_sessions"."date", "coaching_sessions"."timezone", "coaching_sessions"."created_at", "coaching_sessions"."updated_at" FROM "refactor_platform"."coaching_sessions" WHERE "coaching_sessions"."date" > $1"#,
+                [from_date.into()]
+            )]
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn find_by_to_date_returns_all_records_before_date() -> Result<(), Error> {
+        let db = MockDatabase::new(DatabaseBackend::Postgres).into_connection();
+        let mut query_params = HashMap::new();
+        let to_date = NaiveDate::from_ymd_opt(2027, 1, 1).unwrap();
+
+        query_params.insert("to_date".to_owned(), to_date.to_string());
+
+        let _ = find_by(&db, query_params).await;
+
+        assert_eq!(
+            db.into_transaction_log(),
+            [Transaction::from_sql_and_values(
+                DatabaseBackend::Postgres,
+                r#"SELECT "coaching_sessions"."id", "coaching_sessions"."coaching_relationship_id", "coaching_sessions"."date", "coaching_sessions"."timezone", "coaching_sessions"."created_at", "coaching_sessions"."updated_at" FROM "refactor_platform"."coaching_sessions" WHERE "coaching_sessions"."date" < $1"#,
+                [to_date.into()]
+            )]
+        );
+
+        Ok(())
+    }
+}
```

Comment:
One suggestion for another test is to test for a date range that doesn't match any coaching sessions and should return and empty `vec<Model>`. This is something that a user can run into on the frontend, so it'd be good to test for it returning cleanly.

Reply:
We can't really test that here using this methodology as we're not actually testing what the query returns in terms of records. We're testing the SQL generated by the query.

Reply:
More a test for the controller level then? I don't want to get hung up on tests but I did want to ask about anything that came up as a question in my mind while reviewing.